compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -7167,22 +7167,32 @@ private STNode parsePanicKeyword() {
      *
      * <p>
      * <code>
-     * checking-expr := checking-keyword expression
-     * checking-action := checking-keyword action
+     * checking-expr := checking-keyword expression [on-fail-check]
+     * checking-action := checking-keyword action [on-fail-check]
      * </code>
      *
      * @param allowActions Allow actions
      * @param isRhsExpr    Is rhs expression
      * @return Check expression node
      */
     private STNode parseCheckExpression(boolean isRhsExpr, boolean allowActions, boolean isInConditionalExpr) {
+        startContext(ParserRuleContext.CHECKING_EXPRESSION);
```

Comment:
Now that, we start a context here, we need do the same in the errorhandler as well. 
Need add a entry for `CHECKING_EXPRESSION` in `startContextIfRequired()` at errorhandler.